closure/goog/i18n/numberformat.js

```diff
@@ -624,22 +624,104 @@ goog.i18n.NumberFormat.prototype.subformatFixed_ =
   var decimal = goog.i18n.NumberFormatSymbols.DECIMAL_SEP;
   var grouping = goog.i18n.NumberFormatSymbols.GROUP_SEP;
   var zeroCode = goog.i18n.NumberFormat.enforceAsciiDigits_ ?
-                 48  /* ascii '0' */ :
-                 goog.i18n.NumberFormatSymbols.ZERO_DIGIT.charCodeAt(0);
+      48  /* ascii '0' */ :
+      goog.i18n.NumberFormatSymbols.ZERO_DIGIT.charCodeAt(0);
   var digitLen = intPart.length;
+  var nonRepeatedGroupCount = 0;
+  // Keep track of how much has been completed on the non repeated groups
+  var nonRepeatedGroupCompleteCount = 0;
+  var currentGroupSizeIndex = 0;
+  var currentGroupSize = 0;
 
   if (intValue > 0 || minIntDigits > 0) {
     for (var i = digitLen; i < minIntDigits; i++) {
       parts.push(String.fromCharCode(zeroCode));
     }
 
-    for (var i = 0; i < digitLen; i++) {
-      parts.push(String.fromCharCode(zeroCode + intPart.charAt(i) * 1));
+    // If there's more than 1 number grouping,
+    // figure out the length of the non-repeated groupings (on the right)
+    if (this.groupingArray_.length >= 2) {
+      for (var j = 1; j < this.groupingArray_.length; j++) {
+        nonRepeatedGroupCount += this.groupingArray_[j];
+      }
+    }
 
-      if (digitLen - i > 1 && this.groupingSize_ > 0 &&
-          ((digitLen - i) % this.groupingSize_ == 1)) {
-        parts.push(grouping);
+    // Anything left of the fixed number grouping is repeated,
+    // figure out the length of repeated groupings (on the left)
+    var repeatedDigitLen = digitLen - nonRepeatedGroupCount;
+    if (repeatedDigitLen > 0) {
+      // There are repeating digits and non-repeating digits
+      for (var i = 0; i < digitLen; i++) {
+        parts.push(String.fromCharCode(zeroCode + intPart.charAt(i) * 1));
+        if (digitLen - i > 1) {
+          currentGroupSize = this.groupingArray_[currentGroupSizeIndex];
+          if (i < repeatedDigitLen) {
+            // Process the left side (the repeated number groups)
+            var repeatedDigitIndex = repeatedDigitLen - i;
+            // Edge case if there's a number grouping asking for "1" group at
+            // a time; otherwise, if the remainder is 1, there's the separator
+            if (currentGroupSize === 1 ||
+                (currentGroupSize > 0 &&
+                (repeatedDigitIndex % currentGroupSize) === 1)
+            ) {
```

Comment:
I only found it in one place. I used the following regex to try to search for it:
```
^[ ]+\)[ ]*\{
```